### DISCUSSION TITLE
Remap node for vectors (Visual Shader)

### DISCUSSION THREAD

**Discussion by acarx** [godotengine/godot-proposals#9518] [3D]
There's already a remap node for floats, can we get a remap node for vectors?

**Comment by AThousandShips** [comment]
How would this work for vectors? There's no remap on vectors in the scripting, it'd have to exist there first

**Comment by acarx** [comment]
By scripting do you mean shader code?
As far as I know, visual shader nodes work by generating shader code.
Shader code for a vector remap node is the same as float remap, except for 2/3/4 components instead of just one.

**Reply by AThousandShips** [reply]
Sorry meant shaders, there's no `remap` function at all in shaders at the moment, so that's specific to visual shaders

**Reply by acarx** [reply]
Is a remap function necessary? This is how it's implemented for remap node:
```
String VisualShaderNodeRemap::generate_code(Shader::Mode p_mode, VisualShader::Type p_type, int p_id, const String *p_input_vars, const String *p_output_vars, bool p_for_preview) const {
	String code;
	code += "	{\n";
	code += vformat("		float __input_range = %s - %s;\n", p_input_vars[2], p_input_vars[1]);
	code += vformat("		float __output_range = %s - %s;\n", p_input_vars[4], p_input_vars[3]);
	code += vformat("		%s = %s + __output_range * ((%s - %s) / __input_range);\n", p_output_vars[0], p_input_vars[3], p_input_vars[0], p_input_vars[1]);
	code += "	}\n";
	return code;
}
```

Wouldn't it work just as well for vectors (albeit with more lines of generated shader code).

**Comment by tetrapod00** [comment]
Hello from the future! I implemented this, it's in 4.4: https://github.com/godotengine/godot/pull/97314

As this is now implemented, I'm closing this discussion
